### ISSUE TITLE
Add keywords for working with element attributes and values

### ISSUE THREAD

**Issue by spooning** [robotframework/OldSeleniumLibrary#211]
> _Originally submitted to [Google Code](http://code.google.com/p/robotframework-seleniumlibrary/issues/detail?id=211) by mangaroo on 16 Oct 2011_

I may submit a patch with the requested enhancement later on, but in either case, entering issue for tracking purposes and as FYI.

Don't know if it's just me or if element attributes and element values (e.g. input field values) are secondary to text on a page. For me there were some instances where I wanted to verify and wait on element value or some other attribute, and the test framework or library didn't have it built in, so had to patch it myself.

Suggested keyword additions:
- keyword, locator, value or attribute
- Element Value Should Be, locator, value
- Element Attribute Should Be, locator, attribute name, value
- Get Element Value, locator
- Wait Until Page Contains Element Value, locator, value, timeout, error
- Wait Until Page Contains Element Attribute Value, locator, attribute name, value, timeout, error

May not need implement all the suggested keywords, but the more the preferable.

See also <a class="closed_ref" href="https://code.google.com/p/robotframework-seleniumlibrary/issues/detail?id=210" title="Support other locator types for working with element attributes and values"> issue 210 </a> for better way to separate out attribute name from element locator so as not to need to specify as single argument of locator@&#8288;attribute but rather two arguments of locator, attribute.

**Comment by spooning** [comment]
> _Originally submitted to [Google Code](http://code.google.com/p/robotframework-seleniumlibrary/issues/detail?id=211#c1) by @pekkaklarck on 19 Oct 2011_

I close this issue because this covers adding way too many unrelated keywords and some of them are already done. I comment the individual proposals below. Please submit separate issues about the individual keywords that you still think should be added.

1) Element Value Should Be

How would this differ from the existing Element Text Should Be?

2) Element Attribute Should Be

Could be added. You can already now create a user keyword that uses Get Element Attribute and Should Be Equal, though, but we typically have added X Should Be keyword if there is Get X keyword.

3) Get Element Value

There's already Get Value.

4) Wait Until Page Contains Element Value
5) Wait Until Page Contains Element Attribute Value

You can already now implement these utilizing either Wait For Condition or BuiltIn keyword Wait Until Keyword Succeeds. I don't want to add separate Wait Until keywords for all elements because that would just add dozens of really seldom used keywords.

**Comment by spooning** [comment]
> _Originally submitted to [Google Code](http://code.google.com/p/robotframework-seleniumlibrary/issues/detail?id=211#c2) by mangaroo on 19 Oct 2011_

Ok, points understood.

Regarding item 1, Element Value Should Be refers to checking element value of an element like input fields like

&lt;input id="address" value="(enter address here"/&gt;

If I'm not mistaken, Element Text Should Be will only cover text within a element and thus won't work for most form field validation, requiring user to extract the field value using Get Element Attribute, etc. then manually compare with Should Be Equal, etc. keywords.

The only form field that applies to Element Text Should Be is 

&lt;textarea&gt;default text here&lt;/textarea&gt;

And regarding all the above items, ideally as a RF user, I'd prefer to have to implement as few user keywords as needed. And offering useful keywords to the mass is more preferable than to have each group implement their own, even if seldom used, there will still be a minority. However, I also understand as a developer, it makes less sense to implement and maintain less used library keywords.

I guess we'll leave things as is, and reopen if there ever is a request by a larger community besides me for these types of keywords.

**Comment by spooning** [comment]
> _Originally submitted to [Google Code](http://code.google.com/p/robotframework-seleniumlibrary/issues/detail?id=211#c3) by @pekkaklarck on 19 Oct 2011_

Enhancing Element Text Should Be (and other related keywords) so that they handle also input elements would be a good idea. Please submit a separate issue about that!

**Comment by spooning** [comment]
> _Originally submitted to [Google Code](http://code.google.com/p/robotframework-seleniumlibrary/issues/detail?id=211#c4) by mangaroo on 23 Oct 2011_

My mistake for not reviewing clearly the SeleniumLibrary keywords. Element Text Should Be and Element Should Contain don't handle input element values but Textfield Should Contain and Textfield Value Should Be do, so that works out fine. Just a matter of getting used to how the keywords correspond to elements.
